server/src/main/java/org/elasticsearch/index/engine/EngineConfig.java

```diff
@@ -198,6 +203,7 @@ public EngineConfig(
         this.leafSorter = leafSorter;
         this.relativeTimeInNanosSupplier = relativeTimeInNanosSupplier;
         this.indexCommitListener = indexCommitListener;
+        this.recoveryState = recoveryState;
```

Comment:
I wonder if we should capture the `primary` value here instead of the `RecoveryState`, which is volatile and can change? Maybe in a boolean named `recoveredAsPrimary`?

Reply:
I thought it would made more sense to have the full recovery state just in case future engines would need it, but I'm also good with keeping the change minimal :+1:

Reply:
Yes, I agree and that was also my thinking, but for now we only need to differentiate between primary/replica when creating the engine. If more needs come in the future we'll change this